### DISCUSSION TITLE
快点耕博客

### DISCUSSION THREAD

**Discussion by earfanfan** [rbind/yihui#1565] [General]
天啦噜，您那破博客怎么还没耕呢？

**Comment by yihui** [comment]
新工作忙得要死，每天 16 小时的工作强度，上周刷牙看镜子时发现头发都白了一根！（当然也许以前就存在，而我没注意）

不过这工作还有点意思并且很有意义，所以强度是我自找的，要是没意思我就跑路了。今年我计划是集中工作几个月，至少留三个月纯粹做自己的事情（比如写博客）。这是合同工的一个好处，可以凭实力把假期挣出来，自己什么时候觉得一年的粮草挣够了就屁股一拍歇着去，不必全年上班；以前是带薪休假，现在无薪休假，但休假期间天王老子也管不着我了。

**Reply by earfanfan** [reply]
陶渊明说，益辉呀，你是个好孩子，旗帜立得十分鲜明，你这误入尘网中，此去四十年，要学会惜命呐，人类的躯壳可能比你的意志脆弱一些。

从我的视角看，每天16小时的工作强度太高了，恐怕不是长久之计。不过像你这样做事情内驱动力很强又很执着的人，做出这样的选择，倒确实就是你会做的选择。你的朋友我已经学会了课题分离，就不多唠叨啦。不过作为一个才三十出头就颈椎腰椎都不好的人，忍不住想问，你是咋做到连续高强度工作身体不报警的……

另，哈哈，我咋感觉你有点变化，好像语言风格变了一些。总而言之，不觉得累是可以的，累了也可以换换节奏，继续加油吧。

**Reply by yihui** [reply]
这种强度肯定是不能持续的，我也就最多撑三周，然后回归正常。忙主要是在忙打地基。我刚进来的时候一看地基实在是太差了，导致一个团队人人都在浪费时间重复劳动，或瞎忙一些本不用他们操心的事，而且还会彼此拖累。等基础设施建好了，后面我的工作就可以优哉游哉了。这一个多月的经历让我更加意识到软件世界是多么浪费资源，放眼望去，太多高射炮打蚊子的例子。

> 不过作为一个才三十出头就颈椎腰椎都不好的人，忍不住想问，你是咋做到连续高强度工作身体不报警的……

我也母鸡啊。我的腰椎可能是得益于每天三百仰卧起坐、颈椎可能是得益于每周四小时羽毛球吧。感觉你好像更多是在练腿（走路、骑车）。

**Reply by earfanfan** [reply]
有些话不知当讲不当讲（ps益辉说，哈哈，当讲）。你看到的这些问题未必就是基础建设差造成的，但以我现在的水平也说不清楚是什么原因造成的，就像历史上的人事与制度总在变，但是好像有一些历史的规律从未改变，那种东西恐怕就不是小小的益辉能够改变的。你说的团队人浪费时间重复劳动、瞎忙或彼此拖累等问题，虽然我不知道具体是什么情况，但凭直觉觉得应该是人与人之间存在信息壁垒，互相不知道其他人都在做什么或者在一个共同的计划执行过程中对进度没有一个共识，可能你在搭建的基础建设就是用来打破那些信息壁垒的。只不过，好的基础建设也不能打破人心之间的壁垒。

益辉你既是个秩序控又还保留一些理想主义，而且你工作还那么投入，感觉所付出的不仅仅是体力精力还有真心，可是最好也需要明白你能改变什么，以及你不能改变什么。要是将来等你心中的基础设施搭建好了，你看到的那些现象依旧存在，可千万要记住不要把什么事都怪到自己头上啊，不要因外物磨损你的本心。当然啦，你的朋友我衷心地希望你的工作能往你所期待的方向发展，要是有偏的话，哈哈，感觉你的经历说不定也会成为写作的素材呢，现在还是置身事内的视角，等将来置身事外了或许能品尝出些新的味道来。

我下周就恢复打羽毛球的节奏，不过不是公司工会组织的，而是武汉市一些羽毛球俱乐部组织的。一般俱乐部会让每个报名的人自己按照一些规则评估自己的等级，从L1到L8，每级中间还有0.5，我原本以为自己是L2的新手，之前碰到一些L2.5和L3的非新手真是感觉心态被打崩了。唉，感觉如果我能调整好心态努力坚持下去的话，未来几年会有不少进步，我的打球之路还很长，很长……我要加油，不能继续漏油了。

**Reply by yihui** [reply]
我看到的问题没你说的那么深层，目前还是一些触手可及的问题。举个简单例子：大家都用 Git 合作写 R Markdown，但我进去之后发现竟然没有一个机制来保障大家向仓库中推送的更改是可以运行的，大家几乎完全是盲推，也就是不管写了什么就直接推，这会导致错误代码也会被推上去，而下一个人就无法再编译了。最初我喊了好几遍请大家务必在本地编译一遍、确认无误再推送，但响应者寥寥。我本想采用 Github 上的合并请求以及持续集成的工作流，但出于公司内部的种种技术限制无法实现，最后我只好强行为每个人配置了一个 Git 钩子，在推送之前自动编译一遍项目，如果报错就不允许推送。这么一来，世界终于清净了，很难再出现一个人盲目推送的错误代码拖累所有其他合作者的情况。

我这人付出的时候通常不会有多大的期待。事情办得成更好，办不成就算了。对这个新东家，我当然希望能帮到它，尤其是看到领导比较有远见、积极寻求变革时，我会很愿意助他一臂之力。但我最终能起多大作用，还不好说。变革肯定都是有阻力的，大公司的政治会比我的前东家要复杂很多。政治方面的东西是我自己比较厌恶的，也不想去管，只有先把工作做好，往后才有说话的底气。

羽毛球也是台上一分钟、台下几年功的运动，首先起码得让发力水平能到入门级别，也就是高远球能从后场打到对方后场。我前段时间听人讲握力训练的问题，觉得很有道理，你不妨每天练习一下握力，用握力计或什么其它略微难捏的器材，每天快速捏几十次，这同时也会练到小臂肌肉力量。然后就是要领悟，打球的力量是从脚到手指尖的一连串合力，不是单纯手臂力量（尽管手臂力量的比重最大）。

**Reply by earfanfan** [reply]
哲人言，批判的武器不能代替武器的批判。我原先的所思所想是源于本内耗高手从我执中爬出来的经验，哈哈，差点忘了你吃过的盐虽然没我多，走过的路应该比我多，活着的时间也比我长多了，本来就应该很擅长认清自己所处的位置。如果你的改革权力来自于自上而下的赋予，那等你展示出能力以后，自然也能获得自下而上的认同，加油吧，俺干活去也，拜拜。

**Reply by earfanfan** [reply]
我忽然又想唱个反调了……话说俺最近看了不少国家治理与政府行为的书，深刻感受到的一点是，有时候中央的改革政策是出于好意，但执行下来会变得很糟糕。之前看的[一篇文章](http://www.shehui.pku.edu.cn/upload/editor/file/20220309/20220309164004_3717.pdf)有这样一个例子：

>以笔者２０１４年在四川成都进行的 “农民上楼”调查为例。当时的一个研究目的是搞清农民对上楼的看法，分别在上楼前后调查了两次，发现农民在上楼前大都赞成上楼，理由是干净、像城里人；而上楼后则大都说上楼不好，理由是 “贵”、不方便。如果只统计这些态度以及理由，除了罗列一些我们想不到的 “冷知识”之外，对于了解农民的看法没有什么实质的帮助。农民上楼前的 “向往”和上楼后
的烦恼都是真实的，但这些态度是否就是农民的 “心态”？在我们的田野调查更加深入之后，我们有一种奇怪的感受，每当问到这些问题时，农民回答起来似乎过于“随便”，想到什么就说什么。还有个农民间接流露出这样的意思，即 “你们想要什么答案？我按你们的意思回答”。再后来，我们渐渐感受到农民中流行的实际上是这样的心态：无论上楼、下楼，自己既不了解情况，也做不了主，所以随便吧。这是一种 “被上楼”的心态。

变革受到阻力未必就全是因为遭遇了所谓黑暗的职场政治斗争，哈哈，也有可能就是因为变革造成了一种破坏。感觉你那参与变革的经历说不定很精彩，会是很好的写作素材，过几个月俺再来访问后效。

**Reply by yihui** [reply]
并不是什么黑暗的政治斗争，只是我们的工作会动了别人的奶酪，所以人家不愿意为我们提供便利也是完全可以理解的，他们并不是坏人，只是看到新技术来袭之后感到了威胁，这太正常了，就像《铁浆》里镇上的人抗拒火车一样。

**Reply by earfanfan** [reply]
额，要推广提高效率的新技术新工具往往避免不了建立新标准新制度，就像火车要能够开过来也需要先铺设铁轨。这个“所以人家不愿意为我们提供便利也是完全可以理解的”听上去有点沟通不充分的样子，我猜你的新东家里那些被变革的小伙伴们绝不会全都是“感到威胁”的心态，应该也有乐意接受的。哈哈，就让时间去检验吧。

**Reply by ghost** [reply]
个人感觉，R语言的发展自从被tidyverse带偏以后，baseR完全放弃了自主进化，tidyverse的工程化要么生态、性能不行，要么处处要付费，甚至包括rstudio、shiny、quarto都在全面融入python生态。R已经完了……

**Reply by yihui** [reply]
也不能说“带偏”吧，它确实引领了一个新方向或潮流；虽然我至今也不确定这个潮流有几分好、几分坏，但积极的一面肯定是有的。物极必反是普遍规律，好东西一直好到底可能就会开始变坏。我对它的吐槽主要在于生态太臃肿，但我也理解很多人不介意这一点，而更注重语法上的一致性（不过有点讽刺的是，它最容易被吐槽的一个点就是更新迭代太快、版本前后不一致，破坏性的变更总让人要紧追开发者的脚步）。

“处处要付费”也是不存在的，你可能是把 Tidyverse 和 Posit 的商业产品搞混了。Tidyverse 是纯开源免费的，不可能存在付费。

“R 已经完了”更是不知从何说起。至少在我这只青蛙所蹲的这口井里，R 依旧欣欣向荣甚至还在继续腾飞。每当人说什么语言要完的时候，我都想，嗨，想那么多干啥，啥语言到最后不都是要完，当前有口饭吃就好了。人工智能发展到当前这个程度，我觉得已经足以支撑一个掌握了一门程序语言的人成为另一门语言的平庸程序员。就像我完全不懂 Python/Rust/Lua，但有时候想把自己的想法用这些语言实现，问 GPT 很快就能把代码组织出来。所以核心竞争力不在于某种特定程序语言，而是思维、艺术、创意这些软性的东西；至于撸代码，将来大概把大部分工作都交给人工智能去帮忙撸，管它是什么语言。

**Reply by ghost** [reply]
说的有道理，我的视野还是窄了，谢谢指点。关注您的发展，期待新的腾飞……
